ldk/go/examples/search-global/loop/loop.go

```diff
@@ -0,0 +1,83 @@
+package loop
+
+import (
+	"context"
+
+	ldk "github.com/open-olive/loop-development-kit/ldk/go"
+)
+
+const (
+	backgroundColor = "#fff"
+	highlightColor  = "#651fff"
+	primaryColor    = "#666"
+)
+
+// Serve creates the new loop and tells the LDK to serve it
+func Serve() error {
+	l := ldk.NewLogger("loop-example")
+	loop, err := NewLoop(l)
+	if err != nil {
+		return err
+	}
+	ldk.ServeLoopPlugin(l, loop)
+	return nil
+}
+
+// Loop is a structure for generating SideKick whispers
+type Loop struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	sidekick ldk.Sidekick
+	logger   *ldk.Logger
+	style    ldk.Style
+}
+
+// NewLoop returns a pointer to a loop
+func NewLoop(logger *ldk.Logger) (*Loop, error) {
+	return &Loop{
+		logger: logger,
+		style: ldk.Style{
+			BackgroundColor: backgroundColor,
+			HighlightColor:  highlightColor,
+			PrimaryColor:    primaryColor,
+		},
+	}, nil
+}
+
+// LoopStart is called by the host when the plugin is started to provide access to the host process
+func (c *Loop) LoopStart(sidekick ldk.Sidekick) error {
+	c.logger.Info("Starting example controller loop")
+	c.ctx, c.cancel = context.WithCancel(context.Background())
+
+	c.sidekick = sidekick
+
+	return sidekick.UI().ListenGlobalSearch(c.ctx, func(text string, err error) {
+		c.logger.Info("controller loop callback called")
+		if err != nil {
+			c.logger.Error("received error from callback", err)
+			return
+		}
+
+		go func() {
```

Comment:
I guess we'll need to (ideally) take this wrapped goroutine out of the examples when the fix you mentioned is in place, @richseviora? Should I leave a TODO comment here?

Reply:
After some thought, I'm going to leave the Whisper methods as-is. Users will be instructed that the function will be blocking; if they want to close the whisper early they need to cancel the context they're passing in.

So instead I'll be updating the examples so they're all wrapped in a goroutine.

Reply:
Hey @richseviora and @BillyMurrayOlive, I just noticed that when the call to `Whisper().Markdown()` is wrapped in a goroutine for the clipboard example, that the test always passes (I tried putting in some text to make the assertion fail, and it still worked). I assume this is because I need to put in some coordination (or however you say) between the goroutine and the test assertion? Got any ideas about how to do this?

I'm asking because this would probably apply to the tests for the global and searchbar search example loops.

Reply:
Nevermind - I finally grokked just using a channel and blocking on it until it's sent something in the fake Markdown() goroutine. 👍